app/(dashboard)/mailboxes/[mailbox_slug]/saved-replies/savedReplyForm.tsx

```diff
@@ -0,0 +1,180 @@
+        variant: "destructive",
+      });
+    },
+  });
+
+  const deleteSavedReply = api.mailbox.savedReplies.delete.useMutation({
+    onSuccess: () => {
+      toast({ title: "Saved reply deleted successfully", variant: "success" });
+      onDelete?.();
+    },
+    onError: (error) => {
+      toast({ title: "Failed to delete saved reply", description: error.message, variant: "destructive" });
+    },
+  });
+
+  const onSubmit = (data: { name: string; content: string }) => {
+    const finalData = {
+      mailboxSlug,
+      ...data,
+    };
+
+    if (savedReply) {
+      updateSavedReply.mutate({ slug: savedReply.slug, ...finalData });
+    } else {
+      createSavedReply.mutate(finalData);
+    }
+  };
+
+  const handleDelete = () => {
+    if (savedReply) {
+      deleteSavedReply.mutate({ mailboxSlug, slug: savedReply.slug });
+    }
+  };
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-6">
+        <FormField
+          control={form.control}
+          name="name"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Title</FormLabel>
+              <FormControl>
+                <Input placeholder="e.g., Welcome Message" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        <FormField
+          control={form.control}
+          name="content"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Content</FormLabel>
+              <FormControl>
+                <Textarea
+                  placeholder="Enter your saved reply content here..."
+                  className="min-h-32 resize-none"
+                  {...field}
+                />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        <div className="flex items-center justify-between">
+          {savedReply && onDelete ? (
+            <AlertDialog>
+              <AlertDialogTrigger asChild>
+                <Button type="button" variant="destructive_outlined" disabled={deleteSavedReply.isPending}>
+                  Delete
+                </Button>
+              </AlertDialogTrigger>
+              <AlertDialogContent>
+                <AlertDialogHeader>
+                  <AlertDialogTitle>Delete saved reply</AlertDialogTitle>
+                  <AlertDialogDescription>
+                    Are you sure you want to delete "{savedReply.name}"? This action cannot be undone.
+                  </AlertDialogDescription>
+                </AlertDialogHeader>
+                <AlertDialogFooter>
+                  <AlertDialogCancel>Cancel</AlertDialogCancel>
+                  <AlertDialogAction onClick={handleDelete}>Delete</AlertDialogAction>
+                </AlertDialogFooter>
+              </AlertDialogContent>
+            </AlertDialog>
+          ) : (
+            <div />
```

Comment:
Making this

```suggestion
            null
```

Then removing `justify-between` and adding `ml-auto` on the next div instead would be nicer IMO